packages/react-composites/src/composites/CallComposite/utils/Utils.ts

```diff
@@ -188,26 +188,73 @@ export const computeVariant = (
 };
 
 /**
- * Edits the display name of the reomte participant based on the onFetchAvatarPersonaData function given to the call composite
- *
- * @param onFetchAvatarPersonaData Callback function for mutating persona data
- * @param participants Array of remote participants
- * @returns Array of participants with their display name altered
- *
+ * Hook to override the avatar persona data of the users
+ * - Calls {@link AvatarPersonaDataCallback} on each render for provided `userIds`
+ * - Returns an array of the same length as `userIds`. Entries in the array may be undefined if there is no data to return.
  * @private
  */
-export const fetchAvatarPersonaDataAsync = async (
-  onFetchAvatarPersonaData: AvatarPersonaDataCallback,
-  participants: VideoGalleryRemoteParticipant[]
-): Promise<VideoGalleryRemoteParticipant[] | undefined> => {
-  if (onFetchAvatarPersonaData) {
-    await Promise.all(
-      participants.map(async (participant) => {
-        const newParticipantData = await onFetchAvatarPersonaData(participant.userId);
-        participant.displayName = newParticipantData.text ? newParticipantData.text : participant.displayName;
-      })
-    );
-    return participants;
+export const useCustomAvatarPersonaData = (
+  // move this to utils
+  userIds: (string | undefined)[],
+  callBack?: AvatarPersonaDataCallback
+): (AvatarPersonaData | undefined)[] => {
+  const [data, setData] = React.useState<(AvatarPersonaData | undefined)[]>([]);
+  useEffect(() => {
+    (async () => {
+      if (callBack) {
```

Comment:
Use early return to make this more readable.

```
if (!callback) {
  ... default return ...
}
const newData = ...
```

Also, the default return right now is `[]`. We should return a list the same length as `userIds` where all entries are `undefined`.